### PR TITLE
Fix Tracer TSAN Bug

### LINKED ISSUE #13478: Tsan C++ Infrastructure Failures
Example [here](https://sponge.corp.google.com/invocation?id=c7f844df-1277-4592-91ba-9dc8d4c6c05f&searchFor=)

Seems like the problem lies somewhere in here:
```
Traceback (most recent call last):
  File "tools/run_tests/run_tests.py", line 1576, in <module>
    build_only=args.build_only)
  File "tools/run_tests/run_tests.py", line 1487, in _build_and_run
    for language in languages
  File "tools/run_tests/run_tests.py", line 1488, in <genexpr>
    for spec in language.test_specs()
  File "tools/run_tests/run_tests.py", line 348, in test_specs
    stderr=fnull)
  File "/usr/lib/python2.7/subprocess.py", line 573, in check_output
    raise CalledProcessError(retcode, cmd, output=output)
subprocess.CalledProcessError: Command '[u'bins/tsan/bm_call_create', '--benchmark_list_tests']' returned non-zero exit status 66
```

### PR DESCRIPTION
**Pull request by ncteisen**
Fixes #13478

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE              FILE SIZE
 ++++++++++++++ GROWIN ++++++++++++++
  [ = ]       0 [None]    +120  +0.0%

  [ = ]       0 TOTAL     +120  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by matt-kwong** [comment]
Looks like there is a Clang formatting issue. Otherwise, tsan seems to be working!

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE              FILE SIZE
 ++++++++++++++ GROWIN ++++++++++++++
  [ = ]       0 [None]    +120  +0.0%

  [ = ]       0 TOTAL     +120  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by matt-kwong** [comment]
Looks like builds are stuck, restarting them

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE              FILE SIZE
 ++++++++++++++ GROWIN ++++++++++++++
  [ = ]       0 [None]    +120  +0.0%

  [ = ]       0 TOTAL     +120  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by ncteisen** [comment]
Asan #13122 
Tsan #13546

File: src/core/lib/debug/trace.cc [hunk 1]
```diff
@@ -75,8 +75,8 @@ void TraceFlagList::LogAllTracers() {
 }
 
 // Flags register themselves on the list during construction
-TraceFlag::TraceFlag(bool default_enabled, const char* name)
-    : name_(name), value_(default_enabled) {
+TraceFlag::TraceFlag(bool default_enabled, const char* name) : name_(name) {
+  set_enabled(default_enabled);
```
Comment (dgquintas) [inline src/core/lib/debug/trace.cc]: lol

**Pull request merged**